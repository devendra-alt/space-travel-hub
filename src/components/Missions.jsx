import { useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { getMissions } from '../redux/missions/missionsSlice';
import MissionList from './MissionList';
import Header from './Header';


function Mission() {
  // const missionsData = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);

return (
    <>
     
       if (missionsData.loading) {
    return (
      <>
       <Header />
      <div>
        <p>Loading...</p>
      </div>
      </>
    );
  }

  if (missionsData.error) {
    return (
      <>
       <Header />
      <div>
        <p>
          Oops! an error occurred:
          {missionsData.error}
        </p>
      </div>
     </>  
  );
  }
  return (
    <>
    <Header/>
    <div className="container mx-auto px-4">
      <table className="table-auto w-full border-collapse border">
        <thead>
          <tr>
            <th className="border p-2 text-left">Mission</th>
            <th className="border p-2 text-left">Description</th>
            <th className="border p-2 text-left">Status</th>
            <th className="border p-2 text-left">Action</th>
          </tr>
        </thead>
        <tbody>
          {missionsData.missions.map((mission, index) => (
            <MissionList
              key={mission.mission_id}
              mission={mission}
              index={index}
            />
          ))}
        </tbody>
      </table>
    </div>
    </>
  );
}

export default Mission;
