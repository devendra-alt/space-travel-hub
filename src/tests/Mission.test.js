import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'; // Import Provider
import configureStore from 'redux-mock-store'; // Import configureStore
import Missions from '../components/Missions';

// Mock Redux store
const mockStore = configureStore([]);
const initialState = {
  missions: {
    loading: false,
    error: null,
    missions: [
      {
        mission_id: '1',
        mission_name: 'Mock Mission 1',
        description: 'Mock description 1',
        reserved: true,
      },
      {
        mission_id: '2',
        mission_name: 'Mock Mission 2',
        description: 'Mock description 2',
        reserved: false,
      },

    ],
  },
};
const store = mockStore(initialState);

test('Missions component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
