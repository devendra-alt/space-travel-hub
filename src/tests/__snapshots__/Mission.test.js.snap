exports[`Missions component renders correctly 1`] = `
<React.Fragment>
  <Header />
  <div
    className="container mx-auto px-4"
  >
    <table
      className="table-auto w-full border-collapse border"
    >
      <thead>
        <tr>
          <th
            className="border p-2 text-left"
          >
            Mission
          </th>
          <th
            className="border p-2 text-left"
          >
            Description
          </th>
          <th
            className="border p-2 text-left"
          >
            Status
          </th>
          <th
            className="border p-2 text-left"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="border p-4"
          >
            Mock Mission 1
          </td>
          <td
            className="border p-4"
          >
            Mock description 1
          </td>
          <td
            className="border p-4"
          >
            <span
              className="inline-flex items-center rounded-md active-badge whitespace-nowrap"
            >
              ACTIVE MEMBER
            </span>
          </td>
          <td>
            <button
              type="button"
              className="border p-1 leave-button mx-2 rounded whitespace-nowrap"
              onClick={[Function]}
            >
              Leave Mission
            </button>
          </td>
        </tr>
        <tr>
          <td
            className="border p-4"
          >
            Mock Mission 2
          </td>
          <td
            className="border p-4"
          >
            Mock description 2
          </td>
          <td
            className="border p-4"
          >
            <span
              className="inline-flex items-center rounded-md not-member-badge whitespace-nowrap"
            >
              NOT A MEMBER
            </span>
          </td>
          <td>
            <button
              type="button"
              className="border p-1 join-button mx-2 rounded whitespace-nowrap"
              onClick={[Function]}
            >
              Join Mission
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</React.Fragment>
`;
